include/reactor-cpp/assert.hh

```diff
@@ -8,22 +8,28 @@
 
 #pragma once
 
-// A custom definition of assert that mitigates unused variable warnings
-// when assertions are disabled
-#ifdef NDEBUG
-#define ASSERT(x)    \
-  do {               \
-    (void)sizeof(x); \
-  } while (0)
+#ifdef REACTOR_CPP_VALIDATE
+constexpr bool kDoRuntimeValidation = true;
 #else
-#include <cassert>
-#define ASSERT(x) assert(x)
+constexpr bool kDoRuntimeValidation = false;
 #endif
 
+// macro for silencing unused warnings my the compiler
+#define UNUSED(expr) do { (void)(expr); } while (0)
+
+#include <cassert>
 #include <sstream>
 #include <stdexcept>
 #include <string>
 
+constexpr inline void toggle_assert(bool condition) {
+  if constexpr (kDoRuntimeValidation){
```

Comment:
I see assertions and validations as orthogonal. One could debate how much sense this makes, but building with `build-type: Debug` (`#define NDEBUG true`) should include all assertions even if validation is turned off by the user.

Reply:
The changed made should satisfy